libc/src/unistd/linux/getpagesize.cpp

```diff
@@ -0,0 +1,29 @@
+//===-- Implementation of getpagesize -------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/unistd/getpagesize.h"
+#include "src/__support/common.h"
+#include "src/__support/macros/config.h"
+#include "src/__support/macros/page_size.h"
+#include "src/sys/auxv/getauxval.h"
+#include "src/unistd/sysconf.h"
+
+namespace LIBC_NAMESPACE_DECL {
+
+LLVM_LIBC_FUNCTION(int, getpagesize, ()) {
+#if LIBC_PAGE_SIZE == LIBC_PAGE_SIZE_SYSTEM
+  int r = (int)LIBC_NAMESPACE::getauxval(AT_PAGESZ);
+  if (r == 0)
+    return (int)LIBC_NAMESPACE::sysconf(_SC_PAGESIZE);
+  return r;
```

Comment:
Smells a lot like libc/src/unistd/linux/sysconf.cpp...probably this could just be:

```suggestion
  return static_cast<int>(LIBC_NAMESPACE::sysconf(_SC_PAGESIZE));
```

Make sure to use C++ style casts in llvm-libc, please.

Reply:
This is done along with rebasing the PR.